bundles/org.eclipse.core.jobs/src/org/eclipse/core/runtime/jobs/ISchedulableOperation.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Joerg Kubitz.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Joerg Kubitz - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.core.runtime.jobs;
+
+/**
+ * An interface to mark an Operation that needs an ISchedulingRule.
+ *
+ * @since 3.13
+ *
+ */
+public interface ISchedulableOperation {
+	/**
+	 * @return an ISchedulingRule that the operation will acquire - if any. A caller
```

Comment:
The API implies that the task may require a **single** rule **only** or no rule. This is fine if looking on this addition in isolation.

However, the actual change that requests this API (https://github.com/eclipse-platform/eclipse.platform.ui/pull/18) is not necessarily restricted to **one** rule.

I would wait with merging this one and introducing new API if the actual fix may look differently and require different API.

Reply:
There always is the possibility to return a "single" MultiRule. That does not challenge this API.